cmd/fpga_plugin/fpga_plugin_test.go

```diff
@@ -103,6 +108,72 @@ func TestPostAllocate(t *testing.T) {
 	}
 }
 
+func TestAllocate(t *testing.T) {
+	deviceName := "fpga-1"
+
+	request := new(pluginapi.AllocateRequest)
+	creq := new(pluginapi.ContainerAllocateRequest)
+	creq.DevicesIDs = append(creq.DevicesIDs, deviceName)
+	request.ContainerRequests = append(request.ContainerRequests, creq)
+
+	tmpDir, err := os.MkdirTemp("/tmp/", "fpga-allocate")
+	if err != nil {
+		t.Fatal("failed to create temp directory: ", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	dp := &devicePlugin{
+		mode:    regionMode,
+		cdiPath: tmpDir,
+	}
+
+	resp, err := dp.Allocate(request)
+	if resp != nil {
+		t.Errorf("Unexpected non-nil response %+v", resp)
+	}
+
+	if err == nil {
+		t.Error("Unexpected non error")
+	}
+
+	if !errors.Is(err, &dpapi.UseDefaultMethodError{}) {
+		t.Error("not proper error: ", err)
+	}
+
+	content, err := os.ReadFile(filepath.Join(tmpDir, deviceName+".yaml"))
+	if err != nil {
+		t.Errorf("couldn't read CDI device file: %+v", err)
+	}
+
+	ref := cdispec.Spec{
+		Version: CDIVersion,
+		Kind:    CDIKind,
+		Devices: []cdispec.Device{
+			{
+				Name: deviceName,
+				ContainerEdits: cdispec.ContainerEdits{
+					Hooks: []*cdispec.Hook{
+						{
+							HookName: "createRuntime",
+							Path:     "/opt/intel/fpga-sw/intel-fpga-crihook",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	spec := cdispec.Spec{}
+
+	if err = yaml.Unmarshal(content, &spec); err != nil {
+		t.Errorf("failed to unmarshal CDI device spec: %+v", err)
+	}
+
+	if diff := cmp.Diff(ref, spec); len(diff) != 0 {
+		t.Errorf("CDI device spec is invalid:\n%s", diff)
+	}
```

Comment:
Not related to this PR, but it looks like project tests would benefit from using https://pkg.go.dev/github.com/stretchr/testify.

Reply:
Agreed